### DISCUSSION TITLE
Authenticating social media after user created in discord

### DISCUSSION THREAD

**Discussion by CardanoDur** [nextauthjs/next-auth#6916] [Help]
Question for the group, my requirement for an app I'm building is that all signon is done via the Discord provider so that I can do some checks before allowing a person to sign on.  After the user is permitted to sign into the app, I also want to be able to authenticate their social media accounts.  So I allow them to connect to for instance Twitter, using the signin feature after Discord signin, so that I can authenticate their Twitter account.

The way I've started to do this is just having a case statement in the async signOn callback to check account.provider...

`async signIn(params): Promise<string | boolean> {
            const { user, account, profile, email, credentials } = params;

            const client = await clientPromise;
            const db = client.db("mongoDB");

            switch (account!.provider) {
                case "discord":
                    _... check if user is allowed to log in, if so log them in ..._
                case "twitter":
                    _... all I want to do here is get their twitter account, and add it to the user model generated above..._
            }

            return true;`

The issues I'm having are:

1. After the Twitter case, calling the existing session so that I can get the user _id.  The params don't give me the _id key for mongodb that was generated when the user first logged in with Discord.  I add that ID to the session, but I'm unable to call that here in the signIn callback.
2. This method just keeps generating accounts in the accounts collection.  Which leads me to...
3. This might not be the best method to do this.  How are others authenticating social media accounts, not for login, but just for tying a sign-in account to their social media?

**Comment by imyosh** [comment]
Hello 

have you found a solution for this? I'm having the same requirements and after some research I'm surprised no info about this matter, this requirements should be common right!  

I wonder if we just need to implement oauth workflow separately for accounts linking!

**Comment by ladparth** [comment]
Hey there, 

I also have the same question. I am wondering whether you found any solution yet!!

Thanks
